Tipster/Tipster.cpp

```diff
@@ -51,12 +56,72 @@ Tipster::AttachedToWindow()
 {
 	BMessage message(M_CHECK_TIME);
 	fRunner = new BMessageRunner(this, &message, 1000000);
+	
+	BFont linkfont(be_plain_font);
+	linkfont.SetFace(B_UNDERSCORE_FACE);
+	
+	linkStyle.count = 1;
+	linkStyle.runs[0].offset = 0;
+	linkStyle.runs[0].font = linkfont;
+	linkStyle.runs[0].color = make_color(0, 0, 255);
+
+	messenger = new BMessenger(this->Parent());
+	
+	AddBeginningTip();
 
 	BTextView::AttachedToWindow();
 }
 
 
 void
+Tipster::AddBeginningTip()
+{
+	entry_ref ref = GetTipsFile();
+	
+	BFile file(&ref, B_READ_ONLY);
+	if (file.InitCheck() != B_OK)
+		return;
+
+	BString fTip;
+	off_t size = 0;
+	file.GetSize(&size);
+	
+	char* buf = fTip.LockBuffer(size);
+	file.Read(buf, size);
+	fTip.UnlockBuffer(size);
+	
+	BStringList fUpdatedList;
+	fTip.Split("\n", false, fUpdatedList);
+	BString link = fUpdatedList.StringAt(2);
+	
+	Insert(fUpdatedList.StringAt(1));
+	Insert("\n\n<-- Click on the icon to open the URL");
```

Comment:
The whole AddBeginningTip() thing seems to be a bit convoluted. Why not always pick the first tip in the tips-file as the start tip?